tests/TestCase/ORM/RulesCheckerIntegrationTest.php

```diff
@@ -490,6 +490,28 @@ public function testIsUniqueMultipleFieldsOneIsNull()
     }
 
     /**
+     * Tests the IsNotReferencedBy rule before delete
+     *
+     * @group delete
+     * @return void
+     */
+    public function testIsNotReferencedByRule()
+    {
+        $articlesTable = TableRegistry::get('Articles');
+        $articlesTable->belongsTo('Authors');
+        $authorsTable = TableRegistry::get('Authors');
+
+        $author = $authorsTable->get(1);
+        $authorWithoutArticles = $authorsTable->get(2);
+
+        $rules = $authorsTable->rulesChecker();
+        $rules->addDelete($rules->IsNotReferencedBy('Articles'));
```

Comment:
@half2me first make it work with that proxy method - that's probably mandatory to merge (I can't speak for the core).

I do really like configurable binding and foreign keys. They can get you around all the complex situations. It is not that you would need to declare them, only have the option, to. If you don't they are inferred by the association like you already do in your code.